File: binding/src/libqfactoryinterface.cpp
#include <QFactoryInterface>
#include <qfactoryinterface.h>
#include "libqfactoryinterface.h"
#include "libqfactoryinterface.hxx"

QFactoryInterface* QFactoryInterface_new() {
	 return new VirtualQFactoryInterface();
}

QFactoryInterface* QFactoryInterface_new2(const QFactoryInterface* param1) {
	 return new VirtualQFactoryInterface(*param1);
}

libqt_list QFactoryInterface_Keys(const QFactoryInterface* self) {
	return self->keys();
}

void QFactoryInterface_OperatorAssign(QFactoryInterface* self, const QFactoryInterface* param1) {
	self->operator=(*param1);
}

// Base class handler implementation
libqt_list QFactoryInterface_QBaseKeys(const QFactoryInterface* self) {
	auto* vqfactoryinterface = dynamic_cast<const VirtualQFactoryInterface*>(self);
	if (vqfactoryinterface && vqfactoryinterface->isVirtualQFactoryInterface) {
vqfactoryinterface->setQFactoryInterface_Keys_IsBase(true);
	return vqfactoryinterface->keys();
}
}

// Auxiliary method to allow providing re-implementation
void QFactoryInterface_OnKeys(const QFactoryInterface* self, intptr_t slot) {
	auto* vqfactoryinterface = dynamic_cast<const VirtualQFactoryInterface*>(self);
	if (vqfactoryinterface && vqfactoryinterface->isVirtualQFactoryInterface) {
vqfactoryinterface->setQFactoryInterface_Keys_Callback(reinterpret_cast<VirtualQFactoryInterface::QFactoryInterface_Keys_Callback>(slot));
}
}

void QFactoryInterface_Delete(QFactoryInterface* self) {
    delete self;
}

File: binding/src/libqfactoryinterface.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qfactoryinterface.html
pub const qfactoryinterface = struct {

    /// New constructs a new QFactoryInterface object.
    pub fn New() QtC.QFactoryInterface {
        return qtc.QFactoryInterface_new();
    }


    pub fn Keys(self: ?*anyopaque, ) []const u8 {
        return qtc.QFactoryInterface_Keys(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnKeys(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) [*:0]const u8) void {
        qtc.QFactoryInterface_OnKeys(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseKeys(self: ?*anyopaque, ) []const u8 {
        return qtc.QFactoryInterface_QBaseKeys(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QFactoryInterface_Delete(@ptrCast(self));
    }
};
